R/help_create_blogpost.R

```diff
@@ -0,0 +1,160 @@
+library(stringr)
+library(dplyr)
+library(checkmate)
+library(rlang)
+
+
+#' Function to create a new post skeleton
+#'
+#' @param post_name string
+#' @param author string (can be vector for more authors)
+#' @param post_date atomic character date in the format "%Y-%m-%d", e.g. "2023-06-15", defaults to today
+#' @param description blog post description, can be left empty and filled later
+#' @param image one of a set of options for default images
+#' @param tags character vector, some of a set of options
+#'
+#' @return just a message when it worked. Side effect: creates skeleton of blogpost
+#' @export
+#'
+#' @examples
+#' create_post(post_name = "this is my new example post", author = "Stefan Thoma", cover_image = "admiral", tags = "admiral")
+#'
+create_post <- function(post_name,
+                        author = Sys.info()["user"],
+                        post_date = format(Sys.time(), "%Y-%m-%d"),
+                        description = "",
+                        cover_image = available_images,
+                        tags = c(
+                          "metadata",
+                          "submission",
+                          "qc",
+                          "ADaMs",
+                          "SDTMs",
+                          "community",
+                          "conferences",
+                          "admiral",
+                          "roak",
+                          "xportr",
+                          "metatools",
+                          "metacore"
+                        )) {
+
+
+  path_to_img <- "media"
+  available_images <- list.files(path_to_img) %>% tools::file_path_sans_ext()
+
+  # Assert inputs
+  checkmate::matchArg(several.ok = FALSE, cover_image, choices = available_images)
+  checkmate::assert_atomic(post_name)
+  checkmate::assert_atomic(description)
+  checkmate::assert_atomic(cover_image)
+  checkmate::assert_atomic(post_date)
+
+  checkmate::assert_character(post_name)
+  checkmate::assert_character(author)
+  checkmate::assert_character(post_date)
+  if (is.na(as.Date(post_date, format = "%Y-%m-%d"))) {
+    stop('`post_date` has to be in the format "%Y-%m-%d", e.g. "2023-06-15"')
+  }
+
+  checkmate::assert_character(description)
+  checkmate::assert_character(cover_image)
+  checkmate::assert_character(tags)
```

Comment:
I'm wondering if we have admiral loaded with admiraldev if we can just use our assertions rather than checkmate. once less depenedency.

Reply:
Seems to make sense to me. 
But as it's not an admiral blog, admiral.dev is not always loaded. 
Will see if I have time to change it today, otherwise I think its fine to continue without.

Reply:
yes np right now, but I was just thnking about later dependencies.  #57